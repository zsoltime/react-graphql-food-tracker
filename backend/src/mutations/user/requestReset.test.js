'use strict';

const requestReset = require('./requestReset');

describe('requestReset mutation', () => {
  it('updates user with generated token and sends email', async () => {
    const args = { email: 'email@foundindb.io' };

    const mockedContext = {
      db: {
        mutation: { updateUser: jest.fn() },
        query: {
          user: jest.fn().mockImplementation(() => ({ ...args })),
        },
      },
    };
    const res = await requestReset(null, args, mockedContext);

    expect(mockedContext.db.mutation.updateUser).toHaveBeenCalled();
    expect(mockedContext.db.mutation.updateUser).toHaveBeenCalledWith(
      expect.objectContaining({
        data: {
          token: expect.any(String),
          tokenExpiry: expect.any(Number),
        },
        where: { ...args },
      })
    );
    expect(res).toMatchSnapshot();
  });

  it('throws error if email is invalid', () => {
    const args = { email: 'invalidEmail.io' };

    expect(
      requestReset(null, args)
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('throws error if email address is not found in database', () => {
    const args = { email: 'email@notfound.io' };
    const mockedContext = {
      db: {
        query: {
          user: jest.fn().mockImplementation(() => null),
        },
      },
    };

    expect(
      requestReset(null, args, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
    // toMatchSnapshot overwrites previous one for some reason
    expect(mockedContext.db.query.user.mock.calls[0]).toEqual([
      { where: { email: 'email@notfound.io' } },
    ]);
  });
});
