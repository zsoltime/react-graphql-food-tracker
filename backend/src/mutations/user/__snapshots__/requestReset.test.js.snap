// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requestReset mutation throws error if email address is not found in database 1`] = `"Email address not found. You may have entered your email address incorrectly."`;

exports[`requestReset mutation throws error if email is invalid 1`] = `"Email address is invalid"`;

exports[`requestReset mutation updates user with generated token and sends email 1`] = `
Object {
  "message": "Email with token sent",
}
`;
